lib/engine-native/src/artifact.rs

```diff
@@ -545,13 +555,25 @@ impl NativeArtifact {
         })?;
         let metadata_slice: &'static [u8] =
             slice::from_raw_parts(&size[10] as *const u8, metadata_len as usize);
-        let metadata: ModuleMetadata = bincode::deserialize(metadata_slice)
+
+        let mut pos: [u8; 8] = Default::default();
+        pos.copy_from_slice(&metadata_slice[0..8]);
+        let pos: u64 = u64::from_le_bytes(pos);
+        let archived = archived_value::<ModuleMetadata>(&metadata_slice[8..], pos as usize );
+        let mut deserializer = SharedDeserializerAdapter::new(AllocDeserializer);
+        let metadata = RkyvDeserialize::deserialize(archived, &mut deserializer)
```

Comment:
Can we move this into `ModuleMetadata::deserialize_from_archive`? (it will make easier to plug-in other ways of deserialization)